### PR TITLE
Workflow - Python Lint Tests

### LINKED ISSUE #1826: GitHub Actions - Workflow - python X.X lint tests
GitHub Actions workflow for python X.X lint tests.

Quip Doc: https://alteryx.quip.com/J87QAuTDy7pY/Workflow-linttests

### PR DESCRIPTION
**Pull request by ParthivNaresh**
Fixes #1826 

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

**Review by angela97lin — APPROVED** [review]
Nice!! LGTM 🥳

File: .github/workflows/lint_tests.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: Lint Tests
+
+on:
+  pull_request:
+    types: [opened, synchronize]
```
Comment (freddyaboulton) [inline .github/workflows/lint_tests.yml:5]: Does this run after a merge to main?

Comment (ParthivNaresh) [reply .github/workflows/lint_tests.yml:5]: Hard to tell but just to be sure, I'm adding a main branch check

**Review by bchen1116 — APPROVED** [review]
LGTM!

**Review by freddyaboulton — APPROVED** [review]
Nice!

**Pull request merged**